component/organism/AgencyForm/AgencyForm.js

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Button, Input } from "../../atoms";
+import { useForm } from "react-hook-form";
+import styles from "./AgencyForm.module.css";
+
+const AgencyForm = (props) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ reValidateMode: "onSubmit" });
+
+  function submitRequest(e) {
+    // e.preventDefault();
+    console.log(e);
+  }
+  console.log(errors);
+
+  return (
+    <div className={styles.containerMaster}>
+      <form
+        className={styles.container}
+        onSubmit={handleSubmit(async (data, e) => {
+          const reqType = e.nativeEvent.submitter.innerHTML;
+          console.log("data", data);
+          const res = await fetch("api/agencies/agencies", {
+            method: reqType,
```

Comment:
I like the idea that this form could be made reusable for but creating and updating

But here it's better to configure the request method in another way. In this case it should probably be: POST (create)
As the url is api/agencies/

If the request method was PATCH (update)
The url would have to change to api/agencies/:id ( api/agencies/5 for instance ) 

Making this form reusable for both Create and Update can be a bit challinging, maybe a good thing to do together